homeassistant/components/history_stats/data.py

```diff
@@ -110,6 +110,21 @@ async def async_update(
                 )
             )
         ):
+            start_changed = (
+                current_period_start_timestamp != previous_period_start_timestamp
+            )
+            if start_changed:
+                # Trim the start of the _history_current_period to the new period start
+                trim_count = 0
+                for i, state in enumerate(self._history_current_period):
+                    if state.last_changed < current_period_start_timestamp:
+                        state.last_changed = current_period_start_timestamp
```

Comment:
```suggestion
                for i, history_state in enumerate(self._history_current_period):
                    if history_state.last_changed < current_period_start_timestamp:
                        history_state.last_changed = current_period_start_timestamp
```

nit: I thought we mutating ha `State` objects at first

Reply:
Is there a way I could make this clearer?